xo-archiver

```diff
@@ -442,14 +487,89 @@ function copy_to_s3 {
     if aws s3 cp $WORKING_DIR/ $_remote_xva_path/ --recursive
     then
         log "Successfully copied exports of \"$_vm_name\" on bucket \"$AWS_BUCKET_XO_ARCHIVES\""
-        log "Cleaning up $WORKING_DIR/"
-        rm -Rf $WORKING_DIR/
+        clean_working_dir
     else
         die "Fail to push to s3"
     fi
 }
 
 
+# Usage: copy_from_s3 <VM-NAME>
+# copies the <VM-NAME> corresponding files from the s3 bucket specified through envvar
+function copy_from_s3 {
+    if [[ -z $1 ]]
+    then
+        die "copy_from_s3 needs parameter: VM-NAME. E.g: foo-bar-01"
+    fi
+
+    # ensure bucket var is set
+    if [[ -z "$AWS_BUCKET_XO_ARCHIVES" ]]
+    then 
+        die "Bucket is undefined. look at the --help and search for AWS_BUCKET_XO_ARCHIVES"
+    fi
+
+    create_working_dir
+
+    local _remote_xva_path="s3://$AWS_BUCKET_XO_ARCHIVES/$1"
+    log "Copying archives from \"$_remote_xva_path\" into $WORKING_DIR/"
+
+    if aws s3 cp $_remote_xva_path/ $WORKING_DIR/ --recursive --force-glacier-transfer
+    then
+        if ls -A $WORKING_DIR/ | grep -q "[a-Z0-9]"
```

Comment:
I get an error with grep on my local bash: 
```
$ ls -A /tmp | grep -q "[a-Z0-9]"
grep: Invalid range end
```
What am I doing wrong? :thinking:

Reply:
I don't know, `grep` doesn't seems happy, but it should :thinking: 
I can assure you that I multi-tested that and it works both on the prod servers and on my machine